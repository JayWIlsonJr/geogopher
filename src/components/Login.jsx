/*
 * Login.jsx
 */

import React from 'react';
import { NavLink } from 'react-router-dom';

const Login = () => {
  return (
    <div>
      <h1>Login</h1>
      <NavLink exact to="/signup"> sign up </NavLink>
      { /* Login stuff here */ }
    </div>
  );
};

export default Login;
