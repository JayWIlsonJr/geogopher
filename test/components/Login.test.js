/*
 * Login.test.js
 */

import React from 'react';
import Login from '../../src/components/Login';

import renderer from 'react-test-renderer';
import { Link } from 'react-router-dom';
import { StaticRouter } from 'react-router';
import { createRenderer } from 'react-test-renderer/shallow';

global.requestAnimationFrme = function(callback) {
  setTimeout(callback, 0);
};

describe('the login', () => {

  it('renders a Login component', () => {
    const loginRenderer = createRenderer();
    loginRenderer.render(<Login/>);
    let l = loginRenderer.getRenderOutput();
    expect(l.type).toBe('div');
  });

  it('signup link matches snapshot', () => {
    const context = {}
    const tree = require('react-test-renderer').create(
      <StaticRouter location="http://localhost:1337/signup" context={context}>
        <Link to="/signup" />
      </StaticRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });


});
