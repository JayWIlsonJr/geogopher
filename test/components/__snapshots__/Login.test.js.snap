// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the login register link matches snapshot 1`] = `
<a
  href="/register"
  onClick={[Function]}
/>
`;
